File: gerencia_de_estoque/src/main/Empresa.java
package main;

import java.util.ArrayList;

public class Empresa {
	


	private String nome;
	private ArrayList<Filial> filiais;
	
	public Empresa(String nome) {
		this.nome = nome;
		filiais = new ArrayList<Filial>();
	}
	
	public ArrayList<Item> getAllEstoque() {
		return null;
	}

	public void adicionarFilial(String nome, String local, int id) {
		filiais.add(new Filial(nome, local, id));
	}
	
	public void removerFilial(int id) {
		for (Filial filial : filiais) {
			if (filial.getId() == id) {
				filiais.remove(filial);
				break;
			}
		}
	}
	
	public void atualizarFilial(String nome, String local, int id) {
		for (Filial filial : filiais) {
			if (filial.getId() == id) {
				filial.setId(id);
				filial.setLocal(local);
				filial.setNome(nome);
				break;
			}
		}
	}
	
	public String lerFilial() {
		return null;
	}
	
	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public ArrayList<Filial> getFiliais() {
		return filiais;
	}

	public void setFiliais(ArrayList<Filial> filiais) {
		this.filiais = filiais;
	}
	
	public String getFiliaisDescricao() {
		String descricao_filiais = "____FILIAIS____\n\n";
		for (Filial filial : filiais) {
			descricao_filiais += filial.toString();
		}
		return descricao_filiais;
	}
}

File: estoque/app/model/empresa.py
from model.filial import Filial

class Empresa:
    def __init__(self, nome: str):
        self._nome: str = nome
        self._filiais: list[Filial] = []

    def adicionar_filial(self, nome: str, local: str, id: int) -> None:
        nova_filial = Filial(nome, local, id)
        self._filiais.append(nova_filial)

    def buscar_filial(self, id: int) -> Filial | None:
        for filial in self._filiais:
            if filial.id == id:
                return filial
        return None
    
    def remover_filial(self, id: int) -> None:
        self._filiais = [f for f in self._filiais if f.id != id]

    def atualizar_filial(self, nome: str, local: str, novo_id: int, filial_id: int) -> None:
        filial = self.buscar_filial(filial_id)
        if filial:
            filial.nome = nome
            filial.local = local
            filial.id = novo_id